tests/store/artifact/test_sftp_artifact_repo.py

```diff
@@ -196,3 +196,18 @@ def test_delete_selective_artifacts(artifact_path):
         assert not posixpath.exists(posixpath.join(remote_dir, file1))
         assert posixpath.isfile(posixpath.join(remote_dir, directory, file2))
         assert posixpath.isdir(remote_dir)
+
+
+def test_download_sklearn():
+    from mlflow import create_experiment, start_run, sklearn
+
+    with TempDir() as remote:
+        experiment_id = create_experiment(
+            name="sklearn-model-experiment", artifact_location=f"sftp://{remote.path()}"
+        )
+
+        with start_run(experiment_id=experiment_id):
+            original = {"not": "a", "real": "model"}
+            model_uri = sklearn.log_model(original, "model").model_uri
+            downloaded = sklearn.load_model(model_uri)
+            assert downloaded == original
```

Comment:
```
```suggestion
def test_log_and_download_sklearn_model(tmp_path):
    from sklearn.linear_model import LinearRegression

    experiment_id = mlflow.create_experiment(
        name="sklearn-model-experiment", artifact_location=f"sftp://{tmp_path}"
    )
    with mlflow.start_run(experiment_id=experiment_id):
        original = LinearRegression()
        model_uri = mlflow.sklearn.log_model(original, "model").model_uri
        downloaded = mlflow.sklearn.load_model(model_uri)
        assert downloaded == original
```
```

Reply:
I've modified your implementation to keep the dict, `LinearRegression` is not comparable using the `==`-operator so that check always fails. The `mlflow` package wasn't imported in the test case either but aside from those two changes, it's now done in accordance with to your suggested changes.